### PR TITLE
Adds username replacement API

### PR DESCRIPTION
**Pull request by tuchfarber**
New API replaces all copies of username in LMS a desired username. 
Requires user be in username_replacement_admin group. Should only be ran as a larger job  to update usernames across all services,  otherwise the system will be left in a broken state for those users.

### CODE REVIEW

File: ecommerce/extensions/api/v2/views/user_management.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            "successful_replacements": [
+                {"old_username_1": "new_username_1"}
+            ],
+            "failed_replacements": [
+                {"old_username_2": "new_username_2"}
+            ]
+        }
+
+        TODO: Determine if we need an audit trail outside of logging and API response.
```
Comment (bessiesteinberg) [inline ecommerce/extensions/api/v2/views/user_management.py]: remove/resolve TODO

File: ecommerce/extensions/api/v2/views/user_management.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        for obj in post_data:
+            if not (isinstance(obj, dict) and len(obj) == 1):
+                return False
+        return True
+
+    def _replace_username_for_all_models(self, current_username, new_username, replacement_locations):
+        """
+        Replaces current_username with new_username for all (model, column) pairs in replacement locations.
+        Returns if it was successful or not. Will return successful even if no matching
+
```
Comment (bessiesteinberg) [inline ecommerce/extensions/api/v2/views/user_management.py]: not sure what it means by "will return successful even if no matching" does that mean: will return true even if no records were found for 'current_username'.  In what case does it return false?

File: ecommerce/extensions/api/v2/views/user_management.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        TODO: Determine if logs of username are a PII issue.
```
Comment (bessiesteinberg) [inline ecommerce/extensions/api/v2/views/user_management.py]: remove TODO

**Review by bessiesteinberg — APPROVED** [review]
Looks good! There are two TODOs that should be removed or resolved and one comment I thought could use clarification, otherwise the code is really clear and well done. Nice!

File: ecommerce/extensions/api/v2/views/user_management.py (hunk 1 above)
Comment (tuchfarber) [reply ecommerce/extensions/api/v2/views/user_management.py]: Removed

File: ecommerce/extensions/api/v2/views/user_management.py (hunk 2 above)
Comment (tuchfarber) [reply ecommerce/extensions/api/v2/views/user_management.py]: Fixed with text from other PRs. I think I quit typing in the middle of a sentence :). It returns false only if there is an exception while replacing, which should only happen if there's an issue with the DB or the models changes.

File: ecommerce/extensions/api/v2/views/user_management.py (hunk 3 above)
Comment (tuchfarber) [reply ecommerce/extensions/api/v2/views/user_management.py]: Done

File: ecommerce/extensions/api/permissions.py [hunk 4]
```diff
@@ -100,5 +103,13 @@ def has_permission(self, request, view):
                 return False
             enterprise_condition = retrieve_enterprise_condition(coupon)
             enterprise_id = enterprise_condition and enterprise_condition.enterprise_customer_uuid
-
         return self._request_is_permitted_for_enterprise(request, enterprise_id)
+
+
+class CanReplaceUsername(permissions.BasePermission):
+    """
+    Grants access to the Username Replacement API for anyone in the group,
```
Comment (georgebabey) [inline ecommerce/extensions/api/permissions.py]: Same comment as platform - comment mentions group but code just does username check

**Review by georgebabey — APPROVED** [review]
Looks good-  just same comments as the edx-platform PR

**Pull request merged**